src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,100 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
+import { pattern } from '../../api/pattern';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const initialMovieState = {
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  };
+  const [movie, setMovie] = useState<Movie>(initialMovieState);
```

Comment:
Just recomendation create such constants out of component. 
```suggestion
};

const EMPTY_MOVIE = {
    title: '',
    description: '',
    imgUrl: '',
    imdbUrl: '',
    imdbId: '',
  };

export const NewMovie: React.FC<Props> = ({ onAdd }) => {
  const [count, setCount] = useState(0);
  const [movie, setMovie] = useState<Movie>(EMPTY_MOVIE);
```